cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -62,11 +92,125 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 // Update updates a machine and is invoked by the Machine Controller
 func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	glog.Infof("Updating machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+
+	// TODO: Should we lookup by name if this fails?
+	libvirtStatus, err := libvirtutils.ProviderStatus(machine)
+	if err != nil {
+		glog.Errorf("Could not get provider status for update: %v", err)
+		return err
```

Comment:
Why do we, in general, log errors, then return an err? Isn't the caller going to log the error, and perhaps with additional context?

Reply:
Mostly doing this in the "top-level" Update / Create / etc. methods because they are the boundary between the actuator and the upstream controller. I've run into cases where the controller didn't log the returned errors which caused a lot of wasted time debugging.